From 171892a8d92e72515e82751a44b11c3ead4df07a Mon Sep 17 00:00:00 2001
From: Megha Dey <megha.dey@linux.intel.com>
Date: Thu, 13 Sep 2018 16:13:12 -0700
Subject: [PATCH] configure: Prevent unsafe cross-compilation

When cross-compiling, directories from the host machine are included in
EFI_CPPFLAGS. This triggers the following warning from the do_qa_configure
task:

  ERROR: This autoconf log indicates errors, it looked at host include
  and/or library paths while determining system capabilities.

In the autoconf config.log shows clearly the problem:

  cc1: warning: include location "/usr/include/efi" is unsafe for
  cross-compilation [-Wpoison-system-directories]
  cc1: warning: include location "/usr/include/efi/x86_64" is unsafe
  for cross-compilation [-Wpoison-system-directories]

Thus, a "=" is prepended to that the sysroot prefix is added,
as described in the gcc documentation.

Upstream status: applicable only to LUV.

Signed-off-by: Ricardo Neri <ricardo.neri-calderon@linux.intel.com>
Signed-off-by: Megha Dey <megha.dey@linux.intel.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index bce6f0e..7115553 100644
--- a/configure.ac
+++ b/configure.ac
@@ -79,7 +79,7 @@ if test -z "$CRTPATH"; then
    AC_MSG_ERROR([cannot find the gnu-efi crt path])
 fi
 
-EFI_CPPFLAGS="-I/usr/include/efi -I/usr/include/efi/$EFI_ARCH \
+EFI_CPPFLAGS="-I=/usr/include/efi -I=/usr/include/efi/$EFI_ARCH \
  -DEFI_FUNCTION_WRAPPER"
 CPPFLAGS_save="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $EFI_CPPFLAGS"
-- 
1.9.1

